### PR TITLE
Migrate from kubeval to kubeconform

### PR DESCRIPTION
**Pull request by mostafahussein**
This PR will achieve the following:

 - Migrate to [kubeconform](https://github.com/yannh/kubeconform) instead of `kubeval` which is [no longer maintained](https://github.com/instrumenta/kubeval#kubeval).
 - Install `kubeconform` based on CPU architecture e.g. amd64 or arm64
 - The kubeconform tests are limited to the common cpu architectures available from [kubeconform](https://github.com/yannh/kubeconform/releases)
 - Define `kubeconformOpts` as a new optional input for `kubeconform` options, defaults to `-summary`
 - Update `manusa/actions-setup-minikube` action to support ubuntu 22.04 github runners

### CODE REVIEW

**Comment by OliverMKing** [comment]
Thanks for this PR! I'll take a look in the next couple of days (I've been pretty busy).

**Review by OliverMKing — DISMISSED** [review]
Looks good! Thanks for this change. This is a really great update.

**Review by jaiveerk — APPROVED** [review]

**Comment by jaiveerk** [comment]
> @mostafahussein not sure why the Minikube tests aren't working. You might have to revert the version jump on that action.

For context, some of the other actions we maintain use a different setup-minikube action that you may want to try out if you aren't able to get past the failure: https://github.com/Azure/k8s-deploy/blob/main/.github/workflows/run-integration-tests-bluegreen-service.yml#L37

**Review by jaiveerk — DISMISSED** [review]

**Comment by mostafahussein** [comment]
@OliverMKing, will you be able to merge this anytime soon?

**Comment by piroinno** [comment]
@OliverMKing just waiting on you now :-)

**Review by davidgamero — APPROVED** [review]
lgtm

File: .github/workflows/integration-tests.yml [hunk 1]
```diff
@@ -12,11 +12,13 @@ jobs:
       steps:
          - id: setup-minikube
            name: Setup Minikube
-           uses: manusa/actions-setup-minikube@v2.4.2
+           uses: medyagh/setup-minikube@latest
```
Comment (OliverMKing) [inline .github/workflows/integration-tests.yml:15]: Can we pin this to a sha rather than latest? That's pretty important for security.

**Review by davidgamero — APPROVED** [review]

File: .github/workflows/integration-tests.yml (hunk 1 above)
Comment (davidgamero) [reply .github/workflows/integration-tests.yml]: this appears to be resolve in the latest commit

**Comment by mostafahussein** [comment]
hello @OliverMKing , @jaiveerk if you have enough time, is it possible to merge this pr?

**Review by jaiveerk — APPROVED** [review]

**Comment by jaiveerk** [comment]
merged, thanks for this contribution @mostafahussein

**Pull request merged**